### PR TITLE
Add exceptions management

### PR DESCRIPTION
**Pull request by Th3Mouk**

### CODE REVIEW

**Comment by etienneroudeix** [comment]
please fix conflicts

**Comment by Th3Mouk** [comment]
@etienneroudeix ✅

File: src/Rxnet/JsonRpc/Exceptions/UnexpectedServerAdressException.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace Rxnet\JsonRpc\Exceptions;
+
+use Throwable;
+
+class UnexpectedServerAdressException extends RxnetJsonRpcException
+{
+    public function __construct($message = "", $code = 0, Throwable $previous = null)
+    {
+        parent::__construct("Unexpected server address", $code, $previous);
```
Comment (etienneroudeix) [inline src/Rxnet/JsonRpc/Exceptions/UnexpectedServerAdressException.php]: Default string must be in default `$massage` param value

File: src/Rxnet/JsonRpc/Exceptions/InvalidFormatException.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Rxnet\JsonRpc\Exceptions;
+
+use Throwable;
+
+class InvalidFormatException extends RxnetJsonRpcException
+{
+    public function __construct($message = "", $code = 0, Throwable $previous = null)
```
Comment (etienneroudeix) [inline src/Rxnet/JsonRpc/Exceptions/InvalidFormatException.php]: Why redefining `__construct`(same as parent) ?

File: src/Rxnet/JsonRpc/Exceptions/ErrorResponseException.php [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Rxnet\JsonRpc\Exceptions;
+
+use Throwable;
+
+class ErrorResponseException extends RxnetJsonRpcException
+{
+    public function __construct($message = "", $code = 0, Throwable $previous = null)
```
Comment (etienneroudeix) [inline src/Rxnet/JsonRpc/Exceptions/ErrorResponseException.php]: Why redefining `__construct`(same as parent) ?

File: src/Rxnet/JsonRpc/Exceptions/IncorrectResponseException.php [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Rxnet\JsonRpc\Exceptions;
+
+use Throwable;
+
+class IncorrectResponseException extends RxnetJsonRpcException
+{
+    public function __construct($message = "", $code = 0, Throwable $previous = null)
```
Comment (etienneroudeix) [inline src/Rxnet/JsonRpc/Exceptions/IncorrectResponseException.php]: Why redefining `__construct`(same as parent) ?

File: src/Rxnet/JsonRpc/Exceptions/BadStatusCodeException.php [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Rxnet\JsonRpc\Exceptions;
+
+use Throwable;
+
+class BadStatusCodeException extends RxnetJsonRpcException
+{
+    public function __construct($message = "", $code = 0, Throwable $previous = null)
```
Comment (etienneroudeix) [inline src/Rxnet/JsonRpc/Exceptions/BadStatusCodeException.php]: Why redefining `__construct`(same as parent) ?

File: src/Rxnet/JsonRpc/Exceptions/RxnetJsonRpcException.php [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Rxnet\JsonRpc\Exceptions;
+
+use Throwable;
+
+class RxnetJsonRpcException extends \Exception
+{
+    public function __construct($message = "", $code = 0, Throwable $previous = null)
```
Comment (etienneroudeix) [inline src/Rxnet/JsonRpc/Exceptions/RxnetJsonRpcException.php]: Why redefining `__construct`(same as parent) ?

**Review by etienneroudeix — CHANGES_REQUESTED** [review]

**Comment by Th3Mouk** [comment]
@etienneroudeix ✅

**Review by etienneroudeix — CHANGES_REQUESTED** [review]
please update tests in `response.spec.php`to expect specific exceptions

**Comment by etienneroudeix** [comment]
please rebase `master` for ci

**Pull request closed without merging**